lib/persist-state.c

```diff
@@ -237,6 +263,18 @@ _commit_store(PersistState *self)
   return rename(self->temp_filename, self->commited_filename) >= 0;
 }
 
+static gboolean
+_check_watermark(PersistState *self)
+{
+  return (self->current_ofs + PERSIST_FILE_WATERMARK < self->current_size);
+}
+
+static inline gboolean
+_check_free_space(PersistState *self, guint32 size)
+{
+  return (size + sizeof(PersistValueHeader) +  self->current_ofs) <= self->current_size;
```

Comment:
Shouldn't it only be less and not less equal ?

Reply:
I believe less or equal can be allowed here. Equal means exact fit, meaning we can add that last object without expansion.

Less is used when checking the free space with watermark.